packages/fx-core/src/component/driver/script/baseBuildDriver.ts

```diff
@@ -23,18 +41,29 @@ export abstract class BaseBuildDriver extends BaseDeployStepDriver {
 
   async run(): Promise<Map<string, string>> {
     const args = this.toBuildArgs();
-    const commandSuffix = checkMissingArgs("BuildCommand", args.buildCommand).trim();
+    const commandSuffix = checkMissingArgs("BuildCommand", args.args).trim();
     const command = `${this.buildPrefix} ${commandSuffix}`;
+    const env = args.env ? args.env : undefined;
+    await this.progressBar?.start(`Run command ${command} at ${args.workingDirectory}`);
     try {
-      const output = await execute(command, args.src, this.context.logProvider);
-      await this.context.logProvider.debug(`execute ${command} output is ${output}`);
+      const output = await execute(command, args.workingDirectory, this.logProvider, env);
+      await this.logProvider.debug(`execute ${command} output is ${output}`);
+      await this.progressBar?.end(true);
     } catch (e) {
       throw ExecuteCommandError.fromErrorOutput(
         DeployConstant.DEPLOY_ERROR_TYPE,
-        [command, args.src ?? ""],
+        [command, args.workingDirectory ?? ""],
         e
       );
     }
     return BaseBuildDriver.emptyMap;
   }
+
+  /**
+   * call when error happens
+   * do some resource clean up
+   */
+  async cleanup(): Promise<void> {
+    await this.progressBar?.end(false);
```

Comment:
Just curious why here is end with false, but in some files e.g. packages/fx-core/src/component/driver/deploy/azure/azureAppServiceDeployDriver.ts is true?

Reply:
This function is only called when an error occurs.